src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Call.cs

```diff
@@ -451,35 +473,111 @@ private static bool IsSafeForReordering(BoundExpression expression, RefKind kind
             // We start by binding everything that is not obviously reorderable as a temporary, and
             // then run an optimizer to remove unnecessary temporaries.
 
-            BoundExpression[] actualArguments = new BoundExpression[parameters.Length]; // The actual arguments that will be passed; one actual argument per formal parameter.
-            ArrayBuilder<BoundAssignmentOperator> storesToTemps = ArrayBuilder<BoundAssignmentOperator>.GetInstance(rewrittenArguments.Length);
-            ArrayBuilder<RefKind> refKinds = ArrayBuilder<RefKind>.GetInstance(parameters.Length, RefKind.None);
+            // Interpolated string handler conversions introduce another wrinkle to this algorithm, as
+            // they can reuse the receiver or parameters from the call to perform the conversion
+            // as part of the argument list. Our saving grace here is that these conversions _must_
+            // come lexically after the arguments they use, so we don't have to deal with the param
+            // arrays or optional parameters before lowering the conversions. However, because there can
+            // be things such as lambda conversions that are not safe to evaluate twice, we have to skip
+            // the temp optimization steps to ensure that we don't do these double evaluations.
+
+            var actualArguments = new BoundExpression[parameters.Length]; // The actual arguments that will be passed; one actual argument per formal parameter.
+            var storesToTemps = ArrayBuilder<BoundExpression>.GetInstance(arguments.Length);
+            var refKinds = ArrayBuilder<RefKind>.GetInstance(parameters.Length, RefKind.None);
+            var requiresInstanceReceiver = methodOrIndexer.RequiresInstanceReceiver() && methodOrIndexer is not MethodSymbol { MethodKind: MethodKind.Constructor };
```

Comment:
>methodOrIndexer.RequiresInstanceReceiver() && methodOrIndexer is not MethodSymbol { MethodKind: MethodKind.Constructor }; [](start = 43, length = 121)

We have this in a few places. Consider extracting another externsion method.

Reply:
The only two places are in this method. I'm going to leave it as is.